sos/collector/__init__.py

```diff
@@ -619,18 +619,13 @@ def _validate_option(self, default, cli):
                 msg = "Invalid option type for %s. Expected %s got %s"
                 self.exit(msg % (cli.name, default.opt_type, cli.opt_type), 1)
             return cli.value
-        else:
-            val = cli.value.lower()
-            if val not in ['true', 'on', 'yes', 'false', 'off', 'no']:
-                msg = ("Invalid value for %s. Accepted values are: 'true', "
-                       "'false', 'on', 'off', 'yes', 'no'.")
-                self.exit(msg % cli.name, 1)
-            else:
-                if val in ['true', 'on', 'yes']:
-                    return True
-                else:
-                    return False
-        self.exit(f"Unknown option type: {cli.opt_type}")
+
+        val = cli.value.lower()
+        if val not in ['true', 'on', 'yes', 'false', 'off', 'no']:
+            msg = ("Invalid value for %s. Accepted values are: 'true', "
+                   "'false', 'on', 'off', 'yes', 'no'.")
+            self.exit(msg % cli.name, 1)
+        return val in ['true', 'on', 'yes']
```

Comment:
We need to account for non-matching/indeterminate option types here.

Reply:
Actually, looking at this again, perhaps the unknown type is superfluous.

Reply:
yeah, the unknown value is already covered by the previous `if` clause. So the final `self.exit(f"Unknown option type: {cli.opt_type}")` was always an unreachable code.